pkg/registry/pod/rest.go

```diff
@@ -59,6 +59,14 @@ func (podStrategy) ResetBeforeCreate(obj runtime.Object) {
 	}
 }
 
+// PrepareForUpdate clears fields that are not allowed to be set by end users on update.
+func (podStrategy) PrepareForUpdate(obj, old runtime.Object) {
```

Comment:
The podStatusStrategy below embeds podStrategy, so this method will also be called when updating the status. Status would never be updated :\